### DISCUSSION TITLE
SolaX gen 4 apparently not taking settings after upgrade, actually after PIN change

### DISCUSSION THREAD

**Discussion by M0LTE** [wills106/homeassistant-solax-modbus#414] [Q&A] [answered]
Hi
Upgraded my HA and this component yesterday, and now I can't set the values of entities like number.solax_selfuse_nightcharge_upper_soc. A few seconds after changing the value, it just jumps back to the previous value. Using the Unlock Inverter button has no effect.
Cheers
Tom

**Comment by wills106** [comment]
I have added in a new select "Lock State"
Does that show as "Locked" or "Unlocked" now you have pressed the unlock button?
I was going look at removing the buttons as they don't directly return the actual state.

Unless people actually use them in an automation?

Edit: select "Lock State" was introduced in 2023.05.2b1

**Reply by M0LTE** [reply]
sensor.solax_lock_state was disabled, enabled it and the value is Locked

**Reply by M0LTE** [reply]
Ah, I have just seen that I need to upgrade to a beta to get 2023.05.2b1+, doing that now (2b4).

**Reply by M0LTE** [reply]
select.solax_lock_state shows as Locked both before and after pressing button.solax_unlock_inverter

**Reply by wills106** [reply]
If you change the select to "Unlocked" does it retain the value or revert back to "Locked"?
Although either should be able to write to that register.

**Reply by M0LTE** [reply]
It reverts to Locked on the next poll.

Hmm, I have a vague recollection of changing the inverter PIN a few months ago- does the integration require the inverter PIN to be some specific value?

**Reply by wills106** [reply, marked as answer]
Yea it's the default 2014 and 6868 for advanced. You could try "Unlocked - Advanced"

**Reply by M0LTE** [reply]
Yay - changed `select.solax_lock_state` to Unlocked - Advanced, and it has stuck. 
`sensor.solax_lock_state` now has value Unlocked - Advanced.
My changes to things like `number.solax_selfuse_nightcharge_upper_soc` are now sticking.
Thanks and enjoy your ko-fi :-)

**Reply by M0LTE** [reply]
I guess the implication is I changed the default PIN from 2014 to something else, and didn't change the advanced PIN, fair?

**Reply by wills106** [reply]
> Thanks and enjoy your ko-fi :-)

Thanks.

> I changed the default PIN from 2014 to something else, and didn't change the advanced PIN, fair?

For the unlock function to work it needs to send one of the pin codes to the Inverter.
I have never looked at changing either pin, as my Inverter is in the house so no one could change anything on it.

**Reply by M0LTE** [reply]
Right.
I have the pocket wifi, which exposes an open SSID with an open API endpoint on it. I trust it about 1%. Keep meaning to go and pull it out of the inverter.

**Reply by wills106** [reply]
Can you use that pin to change things on the PocketWiFi then?
I have never owned one, so I don't actually know what you can do with the thing, other than I know it's not exactly secure.

**Reply by M0LTE** [reply]
Indirectly yes, via the SolaX Cloud app / website. For setting options using either of those, despite already having authenticated using a username and password, you are required to enter the inverter PIN, and the UI in both the app and the website kind of emulates the physical menu system from the front panel controls. It's really weird.
